extensions/amp-video/0.1/amp-video.js

```diff
@@ -86,8 +85,112 @@ class AmpVideo extends AMP.BaseElement {
     /** @private {!../../../src/mediasession-helper.MetadataDef} */
     this.metadata_ = EMPTY_METADATA;
 
-    /** @private @const {!Array<!UnlistenDef>} */
-    this.unlisteners_ = [];
+    /** @private {?./mixins.AmpVideoMixinDef} */
+    this.mixin_ = null;
+
+    /** @private {?./mixins.PlaybackMixin} */
+    this.playbackMixin_ = null;
+
+    /** @private {boolean} */
+    this.isConnected_ = false;
+  }
+
+  /** @return {boolean} @visibleForTesting */
+  // TODO(alanorozco): Use mediapool service to determine mixin.
+  static isInsideStory(element) {
+    // Small optimization to short-circuit on amp-story-grid-layer instead of
+    // amp-story. Validation defines that all content inside a Story should be
```

Comment:
Well.... sort of 😄 

It has to be in _some_ type of layer; it may be better to check for the class `i-amphtml-story-layer`

Reply:
amp-story-*-layer buildCallback